### PR TITLE
Dsm pepper 4   new permissions

### PR DESCRIPTION
**Pull request by GiCharkviani**
[PEPPER-4](https://broadworkbench.atlassian.net/jira/software/projects/pepper/boards/164?selectedIssue=PEPPER-4)

I have set view permissions according to the requirements in the ticket.
Changes:
-  Added new role ``viewOnlyDSSData``;
- On the participant-page component, added boolean function - `` hasViewOnlyDSSDataPermission `` and applied on tabs accordingly;
- On participant-list component. for dynamic fields, I have added an extra check according to ticket requirements;

**Columns from "Custome View" are deleted in the following way:**
1. They are created at first;
2. And at the end of the data creation flow, runs the function ``removeFieldAccordingToPermission()``, which removes columns according to the requirements mentioned in this ticket

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts [hunk 1]
```diff
@@ -703,6 +704,35 @@ export class ParticipantListComponent implements OnInit {
     });
   }
 
+  private deleteFiltersAccordingToPermission(): void {
+    let columnNamesToDelete: string[];
+
+    if(this.mrAndDssFalse) {
```
Comment (SimoneMaiwald) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts:715]: What about the `t` = tissue columns?

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.html [hunk 2]
```diff
@@ -715,7 +715,7 @@ <h3 class="Color--warn Line--Break">Lost to Follow-Up</h3>
               </ng-container>
             </tab>
 
-          <tab *ngIf="showContactInformation" heading="Contact Information">
+          <tab *ngIf="showContactInformation && (!hasMrViewPermission && hasViewOnlyDSSDataPermission)" heading="Contact Information">
```
Comment (SimoneMaiwald) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.html]: Contact Information still visible for `hasMrViewPermission`, yes? 
If user has `hasMrViewPermission`: Contact Information should be visible

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts [hunk 3]
```diff
@@ -1714,4 +1714,13 @@ export class ParticipantPageComponent implements OnInit, OnDestroy, AfterViewChe
     this.downloading = downloading;
 
   }
+
+  public get hasMrViewPermission(): boolean {
+    return this.role.allowedToViewMedicalRecords();
+  }
+
+  public get hasViewOnlyDSSDataPermission(): boolean {
+    return this.role.viewOnlyDSSData;
```
Comment (SimoneMaiwald) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts:1723]: should that not be `return this.role.viewOnlyDSSData();`

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.ts:1723]: It's getter function, hence we don't need parentheses

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.html (hunk 2 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-page/participant-page.component.html]: Will be resolved! Thank you!

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts (hunk 1 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts:715]: Will be resolved! Thank you!

**Review by SimoneMaiwald — DISMISSED** [review]

**Review by SimoneMaiwald — DISMISSED** [review]

**Review by SimoneMaiwald — APPROVED** [review]

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts [hunk 4]
```diff
@@ -706,6 +708,36 @@ export class ParticipantListComponent implements OnInit {
...
   }
 
+
+  private deleteFiltersAccordingToPermission(): void {
+    let columnNamesToDelete: string[];
+
+    if(this.mrAndDssFalse) {
+      columnNamesToDelete = ['k', 'address', 'm', 'oD'];
+    } else if(this.mrFalseDssTrue) {
+      columnNamesToDelete = ['m', 'oD'];
```
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts:718]: shouldn't tissue be here as well?

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts:718]: we checked that and looks like the tissue is already hidden somewhere on the top

**Pull request merged**